src/main.rs

```diff
@@ -128,8 +130,17 @@ fn main() {
 
         // if list option is given, do not extract
         if matches.is_present("list") {
-            println!("{}", file.name());
-            continue;
+            let result = writeln!(&mut stdout, "{}", file.name());
+            match result {
+                Ok(_) => { continue }
+                Err(err) => {
+                    return if err.kind() == io::ErrorKind::BrokenPipe {
+                        Ok(())
+                    } else {
+                        Err(err)
+                    }
+                }
+            }
```

Comment:
As we only seem to be interested in the error, I think we should only handle the error case:
```suggestion
            if let Err(err)  = writeln!(&mut stdout, "{}", file.name())
            {
                  return if err.kind() == io::ErrorKind::BrokenPipe {
                        Ok(())
                    } else {
                        Err(err)
                    }
            }
            continue;
```

I am not sure I fully understand the intended behaviour: If you receive a `BrokenPipe` you want to exit the program and signal no Error (program completed successfully), and on every other error, exit with an Error?

Reply:
That does look a bit more elegant, thanks! I rushed through the Rust book in order to write my previous PR and I haven't had much of a chance to write Rust since, so I overlooked the `if let` alternative. I guess you could say I'm rusty 😅

Your understanding is correct. My intention was to exit cleanly *only* in the case of a broken pipe. That's a normal situation for a command line program to encounter, and apparently C programs handle that smoothly by default. The reasons for Rust behaving differently are historical, at least according to the comments in the open Rust issue.

I don't know what other IO errors might theoretically occur when writing to stdout — broken pipe seems to be the only one in practice — but it didn't seem prudent to me to quietly ignore them, considering we probably can't log an error message.